### ISSUE TITLE
Migration to MVVM architecture

### ISSUE THREAD

**Issue by rvost** [FlipperPlz/PboExplorer#7]
In PR #6 you mentioned a plan to move to MVVM architecture. I made a small sketch of how this can be done from the state of that PR:

![PboExplorerMVVM drawio](https://user-images.githubusercontent.com/48875452/209365994-57bb4329-dd63-45d2-bf9a-ef84435dcb8a.png)

Pseudo-UML on the left shows how `PboExplorerWindow` state can be split into sub-view-models. Most of the current code belongs to view models layer and can be reused  now and probably refactored later.

Although adoption of MVVM doesn't require any libraries it would be beneficial to bring one to have `INotifyPropertyChange` and `Relay` (aka `Delegate`) commands implemented. I suggest [MVVM Community Toolkit](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/) as it only contains the essentials and can be easily replaced if needed. 

I've picked #6 as a starting point because there will be conflicting changes in `PboExplorerWindow` xaml and code-behinde, also some work of splitting `PboExplorerWindow` into subview already done in that PR. 

If suggestions look good, I can do this migration while you working on BisUtils, but PR need to be merged first.
